scipy/stats/_distn_infrastructure.py

```diff
@@ -1106,7 +1106,7 @@ def rvs(self, *args, **kwds):
             if size == ():
                 vals = int(vals)
             else:
-                vals = vals.astype(int)
+                vals = vals.astype(np.int64)
```

Comment:
I confirmed that this can be an `np.int32` or `np.int64` depending on `sizeof(long) * sizeof(char)` for the system (which is generally 32-bit for 32-bit systems and 64-bit for others). So, best to explicitly convert to `np.int64`.